### PR TITLE
added Trapping Rain Water Algorithm

### PR DESCRIPTION
**Pull request by short-circuit-67**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by short-circuit-67** [comment]
Added the standard stack problem of trapping rain water which can be solved by stack data structure. This can be further used to solve a lot of problems.

File: data_structures/stacks/Trapping_Rain_Water.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	Every element represents the height of the block. The width of the block can be considered 1.
+	Find the volume of trapped rainwater between these block.
+"""
+
+height = [34, 2, 5, 2, 3, 23, 18, 23, 45]
+answer = 162
+
+
+def max_water(height):
+
```
Comment (CaedenPH) [inline data_structures/stacks/Trapping_Rain_Water.py:11]:
```suggestion
```

File: data_structures/stacks/Trapping_Rain_Water.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    """
+    Get the maximum unit of water trapped between the blocks
+
+    >>> max_water(height) == answer
+    True
+
```
Comment (CaedenPH) [inline data_structures/stacks/Trapping_Rain_Water.py:17]:
```suggestion
```

File: data_structures/stacks/Trapping_Rain_Water.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+height = [34, 2, 5, 2, 3, 23, 18, 23, 45]
+answer = 162
+
+
+def max_water(height):
+
+    """
+    Get the maximum unit of water trapped between the blocks
+
+    >>> max_water(height) == answer
```
Comment (CaedenPH) [inline data_structures/stacks/Trapping_Rain_Water.py:15]: You need to create an actual variable for `height` and `answer`. This raised an error, hence why the checks failed

File: data_structures/stacks/Trapping_Rain_Water.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+            answer += distance * min_height
+
+        stack.append(i)
+
+    return answer
+
+
+if __name__ == "__main__":
+
```
Comment (CaedenPH) [inline data_structures/stacks/Trapping_Rain_Water.py:46]:
```suggestion
```

File: data_structures/stacks/Trapping_Rain_Water.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
+"""
+	Every element represents the height of the block. The width of the block can be considered 1.
+	Find the volume of trapped rainwater between these block.
+"""
+
+height = [34, 2, 5, 2, 3, 23, 18, 23, 45]
+answer = 162
```
Comment (CaedenPH) [inline data_structures/stacks/Trapping_Rain_Water.py:7]: These are test inputs, not constants. Move these within the `if __name__ == "__main__":`

File: data_structures/stacks/Trapping_Rain_Water.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
+"""
+	Every element represents the height of the block. The width of the block can be considered 1.
```
Comment (CaedenPH) [inline data_structures/stacks/Trapping_Rain_Water.py:2]:
```suggestion
Every element represents the height of the block. 
The width of the block can be considered 1.
```

**Comment by cclauss** [comment]
Hacktoberfest is strenuous for our team of active volunteer maintainers. That small team now has ___200+ open pull requests___ to review.  We have __high standards for our submissions__ which are described in [`CONTRIBUTING.md`](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md#contributing-guidelines).  Our continuous integration tests help to ensure that our standards are met before we review and merge deserving pull requests.  This is further complicated by the fact that we already have [lots of algorithms](https://github.com/TheAlgorithms/Python/blob/master/DIRECTORY.md) and would like to avoid duplicates.

For these reasons, I will **close all open pull requests failing tests** so that our unpaid team has a smaller mountain to climb.  If you want us to land your pull request, please read [`CONTRIBUTING.md`](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md#contributing-guidelines) _carefully_ and try again.

**Pull request closed without merging**